backend/services/PullRequest.js

```diff
@@ -0,0 +1,66 @@
+const axios = require('axios')
+
+let pEndpoint = (year) => {
+  return `https://api.github.com/search/issues?q=is:pr created:${year}-10-01..${year}-10-31&sort=created&order=desc`
+}
+
+class PullRequest {
+  constructor (year) {
+    this.data = {}
+    this.year = year || (new Date()).getFullYear()
+  }
+  /**
+   * Retrieves all pull requests (github limits them to 100)
+   */
+  async getAll () {
+    try {
+      let response = await axios.get(pEndpoint(this.year))
+
+      this.data = response.data
+
+      return this
+    } catch (error) {
+      throw new Error(error)
+    }
+  }
+
+  /**
+   * Group pull requests by user id
+   */
+  groupByUserId () {
+    let groupedData = {}
+    let data = this.data
+
+    for (let item in data.items) {
+      if (groupedData[data.items[item].user.id]) {
+        groupedData[data.items[item].user.id].push(data.items[item].html_url)
+      } else {
+        groupedData[data.items[item].user.id] = [data.items[item].html_url]
+      }
+    }
+
+    this.data = groupedData
```

Comment:
You can use a [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) here:

```js
// not tested

this.data = this.data.reduce((acc, item) => {
  return { 
    ...acc, 
    [item.user.id]: [...acc[item.user.id], item.html_url]
  }
}, {})
```